### PR TITLE
refactor: update supported frameworks

### PR DESCRIPTION
**Pull request by vbreuss**
Update frameworks in preparation to support [nunit v4](https://docs.nunit.org/articles/nunit/release-notes/framework.html#nunit-400---november-26-2023):
- Update .NET Framework 4.6.1 to .NET Framework 4.6.2
- Remove no longer supported version .NET 5
- Remove .NET Core 3.1 from test project

Related to #1058

BREAKING CHANGE: .NET Framework 4.6.1 and .NET 5 targets are not available anymore because they are [out of support](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core).

### CODE REVIEW

**Review by fgreinacher — CHANGES_REQUESTED** [review]
Very nice, thanks @vbreuss.

Just two minor things I noticed

File: src/System.IO.Abstractions.TestingHelpers/System.IO.Abstractions.TestingHelpers.csproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
       <AssemblyName>System.IO.Abstractions.TestingHelpers</AssemblyName>
       <RootNamespace>System.IO.Abstractions.TestingHelpers</RootNamespace>
       <Description>A set of pre-built mocks to help when testing file system interactions.</Description>
-      <TargetFrameworks>net7.0;net6.0;net5.0;netstandard2.1;netstandard2.0;net461</TargetFrameworks>
+      <TargetFrameworks>net7.0;net6.0;netstandard2.1;netstandard2.0;net462</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (vbreuss) [reply src/System.IO.Abstractions.TestingHelpers/System.IO.Abstractions.TestingHelpers.csproj:6]: I am not sure, why this is displayed this way in GitHub 😕 

In [my branch](https://github.com/vbreuss/System.IO.Abstractions/blob/topic/update-framework-to-462/src/TestableIO.System.IO.Abstractions.TestingHelpers/TestableIO.System.IO.Abstractions.TestingHelpers.csproj) `net8.0` is present and it was [added](https://github.com/TestableIO/System.IO.Abstractions/pull/1032/files#diff-239e928c487ec3a47368cb73c027b87a980da1b5d8d2c23cd090f974a1c64347) in #1032.

Do you have any idea what happened, @fgreinacher?

**Comment by fgreinacher** [comment]
I bump the version to 20.0 and added the breaking change trailer to the description.

**Review by fgreinacher — APPROVED** [review]
Thanks a ton @vbreuss 🙇

**Pull request merged**